### DISCUSSION TITLE
Improve application's accessibility for screenreaders

### DISCUSSION THREAD

**Discussion by Nuno69** [rustdesk/rustdesk#1101] [General]
I would like RustDesk to get much better screenreader accessibility support. I have tested the program and below are my findings
## Windows client
The application is barely usable with a screenreader. I have tested the program with [Zhengdu Screenreader](http://zdsr.com) and the results were much better than with more broadly used [NVDA](https://nvaccess.org). Most problems however are not related to the screenreader in question but to how the interface is being rendered. Iunderstand that Sciter is used as an Ui library in RustDesk,however from what I have read on their site, Microsoft User Interface Automation (UIA) is supported, however the interface needs to be updated appropriately, namely Text Boxes should get a proper label, using the `<label>` HTML tags. The same thing happens inside the file transfering dialog. The lists and text boxeshave no appropriatelabel, but what's more the context menu is not read at all. Moreover there seem to be a few options which are completely unreachable with the keyboard, namely theaddress book and recently connected devices list. They seem to be completely excluded from the TAB navigation order.

## MacOS client

The MacOS client is completely inaccessible, however that seemst o be related to Sciter directly. I post this info here for the sae of completeness...

## mobile clients

Mobile clients are made with Flutter which comes with great support for accessibility by default, howeve rsome stuff have tobe made available by the developer. The problem with Rustdesk mobile clients are buttons which haveonly images as their labels, so screenreaders cannot read the,. Of course, you do not need to add texts to your buttons for all users, Flutter comes with a great "SemanticLabel" widget which can be used for giving graphical controls meaningful names. More about this widget can be read [here](https://api.flutter.dev/flutter/widgets/Icon/semanticLabel.html)
The same technique can be added in order to support accessibility in the web client.

**Comment by rustdesk** [comment]
Yes, not easy to implement, for sure we will do in the future.

**Comment by StormProductionsMusic** [comment]
I can also confirm that the same issue is happening to me using the NVDA screen reader on Windows. I can't even give the password to people that would like to connect to my PC, so therefore I'm completely out of options for now.

**Comment by raivisdejus** [comment]
Minimal set of features needed for RustDesk to be usable as a support tool for blind:
- App navigation with the `tab` key should also focus the One-time password so the person who needs help can hear their password to enable helper to connect
- Keyboard shortcut to switch the sides. With this person who needs help can first connect to the person who can give help and then initiate the side switch so the helper can help.
